FactoryWeather/App/VIP Settings/SettingsInteractor.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  SettingsInteractor.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 09/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import Foundation
+import Promises
+
+protocol SettingsBusinessLogic {
+    func deleteLocation(_ location: Location)
+    func getSavedLocations()
+    func getInitailData()
+    
+    /**
+     Evaluates parameters and according to their state saves settings and calls completion
+     - Parameters:
+        - completion: Completion receives bollean if new weather should be requested
+     */
+    func doneTapped(selectedLocation: Location?, settings: Settings, completion: @escaping ((Bool) -> Void))
```

Comment:
I don't think the naming here is right. The VIP protocols should model high level use-cases, not direct actions on the UI. 

The method name `doneTapped` would relate directly to a UI interaction and not a use-case of saving/persisting the data; which would be a use-case that could be understood regardless of the context of the call site. 

So, an appropriate method name here would be: `save` or something along those lines, that describes the use-case and not the context of what triggered the use-case.

Reply:
`settingsFinished` still refers to the UI flow, not the actual `business logic` use-case. Ask yourself - "What business use-case" needs to occur when the `Done` button's tapped? The `interactor` doesn't care nor should care about the `context` in which it's `business logic` use-case is invoked.